app/src/androidTest/java/com/pajato/argus/ListAdapterTest.kt

```diff
@@ -60,4 +66,39 @@ class ListAdapterTest : ActivityTestBase<MainActivity>(MainActivity::class.java)
         assertTrue("The adapter is not null!", list.adapter == null)
     }
 
+    @Test fun testPersistedData() {
```

Comment:
This looks like it should be in a DatabaseTest.kt file, which will be small.  That file will also include a test for the onUpgrade() function which should be easy: invent some new DB feature, ensure it is not in the current DB, upgrade the DB and ensure the feature is now present in the DB.